### PR TITLE
Audio Function Update

### PR DESCRIPTION
**Pull request by Ollieollieolllie**
Almost ready still testing

### CODE REVIEW

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

**Review by mattysmith22 — COMMENTED** [review]
Asked Brad to take a look first for code style / implementation correctness. After he is happy I will go through

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/cfgFunctions.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#endif
+class CfgFunctions
+{
+	class fza_ah64_project_audioSystem
+	{
+		tag = "FZA_audioSystem";
+		class functions {
+			file = "\fza_ah64_audioSystem\functions";
+			class addASEMessage {R;};
+			class addCaution {R;};
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/cfgFunctions.hpp:14]: How are advisory tones handled? The flight control tone is an example of an advisory tone. There should be a seperate handler for those.

Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/cfgFunctions.hpp:14]: there is no reference to advisory tones in the documents, i need info on what they are, the priority/ implementation for it

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addASEMessage.sqf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+---------------------------------------------------------------------------- */
+params ["_heli", "_priority", "_Reference"];
+
+private _aseMsg = _heli getVariable ["fza_audio_ase_message", ""];
+
+if (_aseMsg isEqualTo "" or {_priority > _aseMsg # 1}) then {
+    _heli setvariable ["fza_audio_ase_message", _this];
+};
+
+private _FuncHook = _heli getVariable ["fza_audio_FuncHook", scriptNull];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addASEMessage.sqf:25]: This should be _funcHook, that way it follows the normal coding conventions already established in the mod.

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addCaution.sqf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    Rosd6(Dryden)
+---------------------------------------------------------------------------- */
+params["_heli"];
+
+_heli setvariable ["fza_audio_caution", true];
+
+private _FuncHook = _heli getVariable ["fza_audio_FuncHook", scriptNull];
+if (scriptDone _FuncHook) then {
+    _FuncHook = [_heli] spawn fza_audioSystem_fnc_audioSystemHandler;
+    _heli setVariable ["fza_audio_FuncHook", _FuncHook];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addCaution.sqf:24]: Please change _FunfHook to _funcHook for both the function name and variable name in lines 23 and 24.

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addWarning.sqf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+---------------------------------------------------------------------------- */
+params ["_heli", "_priority", "_Reference"];
+
+private _wrnMsg = _heli getVariable "fza_audio_warning_message";
+
+if (_wrnMsg == "") then {
+    _heli setvariable ["fza_audio_ase_message", _this];
+};
+
+private _FuncHook = _heli getVariable ["fza_audio_FuncHook", scriptNull];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_addWarning.sqf:25]: Please change _FunfHook to _funcHook for both the function name and variable name in lines 26, 27 and 28

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_audioSystemInit.sqf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+Author:
+    Rosd6(Dryden)
+---------------------------------------------------------------------------- */
+params["_heli"];
+
+_heli setvariable ["fza_audio_ase_message", "", true];
+_heli setvariable ["fza_audio_warning_message", "", true];
+_heli setvariable ["fza_audio_caution", false, true];
+_heli setvariable ["fza_audio_FuncHook", scriptNull, true];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_audioSystemInit.sqf:22]: fza_audio_FuncHook to fza_audio_funcHook

File: @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_audioSystemhandler.sqf [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    7 - ase power 
+
+Parameters:
+    _heli: object - the apache the player is in
+
+Returns:
+    No returns
+
+Examples: 
+    private _FuncHook = _heli getVariable ["fza_audio_FuncHook", scriptNull];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_audioSystem/functions/fn_audioSystemhandler.sqf:20]: Please change _FuncHook to _funcHook...and also fix the lines below as well.

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 7]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
 
 private _Autopage		    = _heli getVariable "fza_ah64_ase_autopage";
 private _Searchlist 		= _heli getVariable "fza_ah64_ase_searchingObj";
-private _aquirelist 		= _heli getVariable "fza_ah64_ase_AquisitionObj";
+private _aquirelist 		= _heli getVariable "fza_ah64_ase_AcquisitionObj";
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:27]: _aquireList should be _acquireList

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 8]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
 private _tracklist 			= _heli getVariable "fza_ah64_ase_trackingobj";
 private _searching     		= [];
-private _aquisition     	= [];
+private _Acquisition     	= [];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:30]: _Acquisition should be _acquisition

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 9]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
-			_heli setVariable ["fza_ah64_ase_AquisitionObj", _aquisition];
-		};
-		if (_Radarstate == "Tracking") then {
-			_tracking pushback _object;
-			_object confirmSensorTarget [playerSide, true];
-			_heli setVariable ["fza_ah64_ase_trackingobj", _tracking];
-		};
-		private _stateAudio = format ["fza_ah64_%1", _Radarstate];
+	// State audio
+	if (_radarState == "Searching") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:47]: Searching should be searching

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 10]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
+		_searching pushback _object;
+		_heli setVariable ["fza_ah64_ase_searchingObj", _searching];
+		_priority = 4;
+	};
+	if (_radarState == "Acquisition") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:52]: Acquisition should be acquisition

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 11]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
+		_Acquisition pushback _object;
+		_heli setVariable ["fza_ah64_ase_AcquisitionObj", _Acquisition];
+		_priority = 5;
+	};
+	if (_radarState == "Tracking") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:57]: Tracking should be tracking

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 12]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
-			[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
-		};
-		if (_Autopage == 2 && _Radarstate == "Aquisition") then {
-			[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
-		};
-		if (_Autopage == 3 && _Radarstate == "Tracking") then {
-			[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
-		};
+	//ASE AUTOPAGE
+	if (_Autopage == 1 && _radarState == "Searching") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:74]: Searching should be searching

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 13]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
+		[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
+	};
+	if (_Autopage == 2 && _radarState == "Acquisition") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:77]: Acquisition should be acquisition

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 14]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
+		[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
+	};
+	if (_Autopage == 3 && _radarState == "Tracking") then {
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:80]: Tracking should be tracking

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf [hunk 15]
```diff
@@ -24,67 +24,64 @@ if !(isEngineOn _heli) exitwith {};
...
+		[_heli, 1, "ase"] call fza_mpd_fnc_setCurrentPage;
 	};
 } foreach _audioList;
 
 _heli setVariable ["fza_ah64_ase_searchingObj", _searching];
-_heli setVariable ["fza_ah64_ase_AquisitionObj", _aquisition];
+_heli setVariable ["fza_ah64_ase_AcquisitionObj", _Acquisition];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseAudioController.sqf:86]: AcquisitionObj should be acquisitionObj and Acquisition should be acquisition.

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseRLWR.sqf [hunk 16]
```diff
@@ -49,7 +49,7 @@ if (_rlwrPwr == "on") then {
         if ("marked" in _type) then {
             if (_distance_m <= ASE_ACQ_RANGE_M) then {
                 _rlwrObjects pushBack [ASE_ACQ, _heli getRelDir _object];
-                _rlwrAudio pushback [vehicle _object, "Aquisition"];
+                _rlwrAudio pushback [vehicle _object, "Acquisition"];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ASE/fn_aseRLWR.sqf:52]: Acquisition should be acquisition

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/damage/fn_damageM230.sqf [hunk 17]
```diff
@@ -86,7 +86,7 @@ if (_heatfactor > 1200) then {
 _rand = random _heatfactor;
 
 if (_rand > _heatchance) then {
-    //["fza_ah64_bt_gun", 0.5, "fza_ah64_bt_jammed", 1] spawn fza_fnc_playAudio;
+    //Mastercaution here
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/damage/fn_damageM230.sqf:89]: When the gun fails there is no alert to the crew via audio. The WPN page will display FAIL on the gun and the gun will be colored yellow.

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/damage/fn_damageSystem.sqf [hunk 18]
```diff
@@ -36,28 +36,31 @@ if (!(alive _heli)) exitwith {
 
 if (_system == "mala vrtule") then {
     if (_usesound && _oldDam < 0.4 && _damage >= 0.4) then {
-        ["fza_ah64_tail_rotor_damaged_1", 3] spawn fza_fnc_playAudio;
+        //Mastercaution here
+    };
+    if (_usesound && _oldDam < 0.8 && _damage >= 0.8) then {
+        [_heli, 2, "EngineWarning", "fza_ah64_tail_rotor_hydraulic_failure", 2] call fza_audioSystem_fnc_addASEMessage;
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/damage/fn_damageSystem.sqf:42]: EngineWarning should be engineWarning

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/event/fn_eventInit.sqf [hunk 19]
```diff
@@ -82,7 +82,7 @@ if (!(_heli getVariable ["fza_ah64_aircraftInitialised", false]) && local _heli)
     //--Autopage    - Crewstation independent
     _heli setVariable ["fza_ah64_ase_autopage",     1];
     _heli setVariable ["fza_ah64_ase_searchingObj",  []];
-    _heli setVariable ["fza_ah64_ase_AquisitionObj",  []];
+    _heli setVariable ["fza_ah64_ase_AcquisitionObj",  []];
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/event/fn_eventInit.sqf:85]: AcquisitionObj should be acquisitionObj

File: @AH-64D Apache Longbow/Addons/fza_ah64_mpd/functions/page/fn_pageASEHandleControl.sqf [hunk 20]
```diff
@@ -19,7 +19,7 @@ switch(_control) do {
 		private _newState = ["off", "on"] select (_heli getVariable "fza_ah64_ase_rlwrPwr" == "off");
         _heli setVariable ["fza_ah64_ase_rlwrPwr", _newState];
 		if (_newstate == "on") then {
-			["fza_ah64_APR_39_Power_up", 1.8] spawn fza_fnc_playAudio;
+			[_heli, 7, "APR-39", "fza_ah64_APR_39_Power_up", 1.8] call fza_audioSystem_fnc_addASEMessage;
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_mpd/functions/page/fn_pageASEHandleControl.sqf:22]: ...Power_up should be power_up

File: @AH-64D Apache Longbow/Addons/fza_ah64_mpd/functions/page/fn_pageENGDraw.sqf [hunk 21]
```diff
@@ -56,7 +56,8 @@ _heli setUserMFDText [MFD_INDEX_OFFSET(MFD_TEXT_IND_ENG_NR), _rotorRpm toFixed 0
 
 //TODO: Change so sound occurs even if not in engine page
 if (_rotorRpm > 110 && isengineon _heli && (getpos _heli select 2) > 5 && !fza_ah64_warnHighRpm) then {
-    ["fza_ah64_rotor_rpm_high", 1] spawn fza_fnc_playAudio;
+	[_heli, 1, "EngineWarning", "fza_ah64_rotor_rpm_high", 1.5] call fza_audioSystem_fnc_addASEMessage;
```
Comment (BradMick) [inline @AH-64D Apache Longbow/Addons/fza_ah64_mpd/functions/page/fn_pageENGDraw.sqf:59]: EngineWarning should be engineWarning

**Review by BradMick — COMMENTED** [review]
Please review the comments provided and make the requested changes.

**Review by mattysmith22 — DISMISSED** [review]
Good job!

**Review by mattysmith22 — APPROVED** [review]
Looks good!

**Pull request merged**